### PR TITLE
Provide Nadel hook for applying a schema transformation

### PR DESCRIPTION
**Pull request by danieltreacy**
Extends the Nadel builder to accept a new `SchemaTransformation`
interface that accepts a graphql schema object, and returns a new schema
object.

### CODE REVIEW

File: src/main/java/graphql/nadel/schema/SchemaTransformation.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ *             }
+ *         }
+ *     }
+ * </code>
+ *
+ * @see graphql.schema.SchemaTransformer
+ * @see graphql.schema.GraphQLTypeVisitorStub
+ */
+@PublicSpi
+public interface SchemaTransformation {
```
Comment (bbakerman) [inline src/main/java/graphql/nadel/schema/SchemaTransformation.java]: Naming is hard.  Can we consider calling this `SchemaTransformationHook`

Its like the other hook we have and its a callback. We can start to name out SPI callbacks "hooks" say

This is confusingly named to the underlying SchemaTransformation in graphql-java.

Comment (danieltreacy) [reply src/main/java/graphql/nadel/schema/SchemaTransformation.java]: Yeah that makes sense. Will rename.

Comment (andimarek) [reply src/main/java/graphql/nadel/schema/SchemaTransformation.java]: 👍  for the rename

File: src/main/java/graphql/nadel/schema/SchemaTransformationHook.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
+package graphql.nadel.schema;
+
+import graphql.PublicSpi;
+import graphql.schema.GraphQLSchema;
+
+/**
+ * High level representation of a a graphql schema transformation. Warning: the schema resulting from the transformation
```
Comment (andimarek) [inline src/main/java/graphql/nadel/schema/SchemaTransformationHook.java]: Maybe make clear that this is actually the overall schema that is transformed, not the underlying one.

File: src/main/java/graphql/nadel/Nadel.java [hunk 3]
```diff
@@ -133,10 +137,14 @@ private GraphQLSchema createOverallSchema() {
                 .map(Service::getDefinitionRegistry)
                 .collect(toList());
         GraphQLSchema schema = overallSchemaGenerator.buildOverallSchema(registries, commonTypes, overallWiringFactory);
+
+        // apply transformations
```
Comment (andimarek) [inline src/main/java/graphql/nadel/Nadel.java]: we don't really need this comment :-)

Comment (danieltreacy) [reply src/main/java/graphql/nadel/Nadel.java]: Removed

File: src/main/java/graphql/nadel/schema/SchemaTransformationHook.java (hunk 2 above)
Comment (danieltreacy) [reply src/main/java/graphql/nadel/schema/SchemaTransformationHook.java]: Done.

**Pull request merged**